win32ss/user/winsrv/consrv/condrv/text.c

```diff
@@ -977,6 +977,18 @@ IntWriteConsoleOutputStringUnicode(
         if (bCJK && Ptr->Char.UnicodeChar >= 0x80 &&
             mk_wcwidth_cjk(Ptr->Char.UnicodeChar) == 2)
         {
+            if (X == Buffer->ScreenBufferSize.X - 1)
+            {
+                /* new line */
+                X = 0;
+                ++Y;
+                if (Y == Buffer->ScreenBufferSize.Y)
```

Comment:
Hmm Now that I see that code (and the similar code patterns I originally wrote there) I think that there may be somewhat buggy. I will have a look at these separately later once your PR is merged.